pandas/core/frame.py

```diff
@@ -4166,11 +4167,13 @@ def isetitem(self, loc, value) -> None:
         self._iset_item_mgr(loc, arraylike, inplace=False, refs=refs)
 
     def __setitem__(self, key, value) -> None:
-        if not PYPY and using_copy_on_write():
-            if sys.getrefcount(self) <= 3:
+        if not PYPY:
+            if using_copy_on_write() and sys.getrefcount(self) <= 3:
                 warnings.warn(
                     _chained_assignment_msg, ChainedAssignmentError, stacklevel=2
                 )
+            elif warn_copy_on_write() and sys.getrefcount(self) <= 3:
+                warnings.warn("ChainedAssignmentError", FutureWarning, stacklevel=2)
```

Comment:
Is the missing message intentional?

Reply:
No, just WIP, was focusing on the mechanics first